napari/plugins/__init__.py

```diff
@@ -1,34 +1,49 @@
+from functools import lru_cache
```

Comment:
```suggestion
from functools import cache
```

lru have extra , slower logic to evic items, that you don't need if you know you have a small limited set of output.